test/zod-collection.test.ts

```diff
@@ -1,34 +1,58 @@
 import { z } from 'zod'
-import { convertToZodCollection } from '../src'
+import { convertToZodCollection, TsReadWriteCollection, WithTime } from '../src'
 import { closeDb, mkTsTestCollection } from './util'
+import { ObjectId, WithId } from 'mongodb'
 
 const ZExample = z.object({
   a: z.number(),
   b: z.string(),
   numbers: z.number().array(),
 })
+interface ZExample extends z.infer<typeof ZExample> {}
 
 const mkZodCollection = async () => {
-  const col = await mkTsTestCollection()
-  // use any here so we can insert wrong values without upsetting typescript
-  const zodCol = convertToZodCollection<any>(col, ZExample)
-  return zodCol
+  const col = await mkTsTestCollection<WithTime<ZExample>>()
+  return convertToZodCollection(col, ZExample)
 }
 
 test('Documents should be parsed correctly on insert', async () => {
   const zodCol = await mkZodCollection()
 
   expect(() =>
+    // @ts-expect-error
     zodCol.insertOne({ a: 'string', b: 'string', numbers: [9] })
   ).toThrow()
   await expect(
     zodCol.insertOne({ a: 0, b: 'string', numbers: [0] })
   ).resolves.toBeTruthy()
+})
 
-  expect(() => zodCol.insertMany([{ a: 0, b: 0 }])).toThrow()
-  await expect(
-    zodCol.insertMany([{ a: 0, b: 'string', numbers: [7] }])
-  ).resolves.toBeTruthy()
```

Comment:
We should keep / add a test that uses `insertMany` just to ensure the validation works there as well

Reply:
Thanks for noticing! Removing this was a mistake